src/rlookup.c

```diff
@@ -264,6 +264,42 @@ static RSValue *hvalToValue(RedisModuleString *src, RLookupCoerceType type) {
   }
 }
 
+static RSValue *jsonValToValue(RedisModuleCtx *ctx, RedisJSON json) {
+  size_t len;
+  char *str;
+  const char *constStr;
+  RedisModuleString *rstr;
+  RSValue *rs_val;
+  long long ll = 0;
+  double dd = 0.0;
+  int i;
+
+  switch (japi->getType(json)) {
+    case JSONType_String:
+      japi->getString(json, &constStr, &len);
+      str = rm_strndup(constStr, len);
+      return RS_StringVal(str, len);
+    case JSONType_Int:
+      japi->getInt(json, &ll);
+      return RS_Int64Val(ll);
+    case JSONType_Double:
+      japi->getDouble(json, &dd);
+      return RS_NumVal(dd);
+    case JSONType_Bool:
+      japi->getBoolean(json, &i);
+      return RS_Int64Val(i);
+    case JSONType_Array:
+    case JSONType_Object:
+      japi->getJSON(json, ctx, &rstr);
+      rs_val = RS_RedisStringVal(rstr);
+      return rs_val;
```

Comment:
`japi->get*` functions can fail, but we don't have a good way to report the error. Still, completely ignoring the return value is problematic.